modules/gltf/editor/editor_scene_exporter_gltf_settings.cpp

```diff
@@ -0,0 +1,166 @@
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "editor_scene_exporter_gltf_settings.h"
+
+const uint32_t PROP_EDITOR_SCRIPT_VAR = PROPERTY_USAGE_EDITOR | PROPERTY_USAGE_SCRIPT_VARIABLE;
+
+bool EditorSceneExporterGLTFSettings::_set(const StringName &p_name, const Variant &p_value) {
+	String name_str = String(p_name);
+	if (name_str.contains("/")) {
+		return _set_extension_setting(name_str, p_value);
+	}
+	if (p_name == StringName("image_format")) {
+		_document->set_image_format(p_value);
+		emit_signal("property_list_changed");
+		return true;
+	}
+	if (p_name == StringName("lossy_quality")) {
+		_document->set_lossy_quality(p_value);
+		return true;
+	}
+	return false;
+}
+
+bool EditorSceneExporterGLTFSettings::_get(const StringName &p_name, Variant &r_ret) const {
+	String name_str = String(p_name);
+	if (name_str.contains("/")) {
+		return _get_extension_setting(name_str, r_ret);
+	}
+	if (p_name == StringName("image_format")) {
+		r_ret = _document->get_image_format();
+		return true;
+	}
+	if (p_name == StringName("lossy_quality")) {
+		r_ret = _document->get_lossy_quality();
+		return true;
+	}
+	return false;
+}
+
+void EditorSceneExporterGLTFSettings::_get_property_list(List<PropertyInfo> *p_list) const {
+	for (PropertyInfo prop : _property_list) {
+		if (prop.name == "lossy_quality") {
+			String image_format = get("image_format");
+			bool is_image_format_lossy = image_format == "JPEG" || image_format.findn("Lossy") != -1;
```

Comment:
we should document this "Lossy" substring behavior on `get_saveable_image_formats()`

Reply:
Done, documentation updated. `If the format name contains [code]"Lossy"[/code], the lossy quality slider will be displayed.`